### PR TITLE
build controller with multi stage

### PR DESCRIPTION
**Pull request by hchenxa**
**What this PR does / why we need it**:

build the api-controller with multi stage.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -161,17 +163,11 @@ manifests:
...
-openstack-cluster-api-controller: depend manager manifests
-ifeq ($(GOOS),linux)
-	cp bin/manager cmd/manager
-	docker build -t $(REGISTRY)/openstack-cluster-api-controller:$(VERSION) cmd/manager
-	rm cmd/manager/manager
-else
-	$(error Please set GOOS=linux for building the image)
-endif
+openstack-cluster-api-controller: generate fmt vet manifests
+	docker build . -f cmd/manager/Dockerfile --network=host -t "$(REGISTRY)/openstack-cluster-api-controller:$(VERSION)"
```
Comment (gyliu513) [inline Makefile]: Why ` --network=host`?

Comment (hchenxa) [reply Makefile]: the docker image required to install package for apt repo, and the docker image bydefault use the default network to build docker image which may not able to access the internet.

Comment (gyliu513) [reply Makefile]: If the container cannot access internet, then the host may not able to access internet either?

Comment (hchenxa) [reply Makefile]: there should be the DNS problem, the host can solve the deb.debian.org but the container can not.

here is the message from containers:
```
Err:1 http://deb.debian.org/debian stretch InRelease
  Temporary failure resolving 'deb.debian.org'
Err:2 http://security.debian.org/debian-security stretch/updates InRelease
  Temporary failure resolving 'security.debian.org'
```

Comment (gyliu513) [reply Makefile]: FYI https://stackoverflow.com/questions/27435479/pass-net-host-to-docker-build

/approve

**Pull request merged**